typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,59 +1,691 @@
+    }
+
+    const ismType = config.type;
+    const logger = rootLogger.child({ chainName: this.chainName, ismType });
+
+    logger.debug(`Deploying ${ismType} to ${this.args.chain}`);
+
+    switch (ismType) {
+      case IsmType.MESSAGE_ID_MULTISIG:
+      case IsmType.MERKLE_ROOT_MULTISIG:
+        return this.deployMultisigIsm({
+          config,
+          logger,
+        });
+
+      case IsmType.ROUTING:
+      case IsmType.FALLBACK_ROUTING:
+        return this.deployRoutingIsm({
+          config,
+          logger,
+        });
+
+      case IsmType.AGGREGATION:
+        return this.deployAggregationIsm({
+          config,
+          logger,
+        });
+
+      case IsmType.OP_STACK:
+        return this.deployer.deployContractFromFactory(
+          this.chainName,
+          new OPStackIsm__factory(),
+          IsmType.OP_STACK,
+          [config.nativeBridge],
+        );
+
+      case IsmType.PAUSABLE:
+        return this.deployer.deployContractFromFactory(
+          this.chainName,
+          new PausableIsm__factory(),
+          IsmType.PAUSABLE,
+          [extractOwnerAddress(config.owner)],
+        );
+
+      case IsmType.TRUSTED_RELAYER:
+        assert(
+          this.args.addresses.mailbox,
+          `Mailbox address is required for deploying ${ismType}`,
+        );
+        return this.deployer.deployContractFromFactory(
+          this.chainName,
+          new TrustedRelayerIsm__factory(),
+          IsmType.TRUSTED_RELAYER,
+          [this.args.addresses.mailbox, config.relayer],
+        );
+
+      case IsmType.TEST_ISM:
+        return this.deployer.deployContractFromFactory(
+          this.chainName,
+          new TestIsm__factory(),
+          IsmType.TEST_ISM,
+          [],
+        );
+
+      default:
+        throw new Error(`Unsupported ISM type ${ismType}`);
+    }
+  }
+
+  protected async deployMultisigIsm({
+    config,
+    logger,
+  }: {
+    config: MultisigIsmConfig;
+    logger: Logger;
+  }): Promise<IMultisigIsm> {
+    const signer = this.multiProvider.getSigner(this.chainName);
+    const factoryName =
+      config.type === IsmType.MERKLE_ROOT_MULTISIG
+        ? 'staticMerkleRootMultisigIsmFactory'
+        : 'staticMessageIdMultisigIsmFactory';
+
+    const address = await EvmIsmModule.deployStaticAddressSet({
+      chain: this.chainName,
+      factory: this.factories[factoryName],
+      values: config.validators,
+      logger,
+      threshold: config.threshold,
+      multiProvider: this.multiProvider,
+    });
+
+    const contract = IMultisigIsm__factory.connect(address, signer);
+
+    // TODO: figure out how to get the constructor arguments for manual deploy TXs
+    // const verificationInput = buildVerificationInput(
+    //   proxyFactoryImplementations[factoryName],
+    //   contract.address,
+    //   constructorArgs,
+    // );
+    // await this.deployer.verifyContract(
+    //   this.chainName,
+    //   verificationInput,
+    //   logger,
+    // );
```

Comment:
I think you can pull in the verify logic to the `deployStaticAddressSet` helper

Reply:
leaving a comment to verify in the helper